util/test/demos/gl/gl_backdoor_context.cpp

```diff
@@ -0,0 +1,178 @@
+/******************************************************************************
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019-2020 Baldur Karlsson
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ ******************************************************************************/
+
+#include "gl_test.h"
+#include <time.h>
+
+RD_TEST(GL_Backdoor_Context, OpenGLGraphicsTest)
+{
+  static constexpr const char *Description =
+      "Given an application with contexts A and B (not shared). Starts the capture with context A, "
+      "then activates context B mid-frame and renders using some resources that are deferred, then "
+      "activates context A again before the end of the frame";
+
+  std::string common = R"EOSHADER(
+
+#version 420 core
+
+#define v2f v2f_block \
+{                     \
+	vec4 pos;           \
+	vec4 col;           \
+	vec4 uv;            \
+}
+
+)EOSHADER";
+
+  std::string vertex = R"EOSHADER(
+
+layout(location = 0) in vec3 Position;
+layout(location = 1) in vec4 Color;
+layout(location = 2) in vec2 UV;
+
+out v2f vertOut;
+
+void main()
+{
+	vertOut.pos = vec4(Position.xyz, 1);
+	gl_Position = vertOut.pos;
+	vertOut.col = Color;
+	vertOut.uv = vec4(UV.xy, 0, 1);
+}
+
+)EOSHADER";
+
+  std::string pixel = R"EOSHADER(
+
+in v2f vertIn;
+
+layout(location = 0, index = 0) out vec4 Color;
+
+layout(binding = 0) uniform sampler2D tex;
+
+void main()
+{
+	Color = vertIn.col * texture(tex, vertIn.uv.xy);
+}
+
+)EOSHADER";
+
+  int main()
+  {
+    // initialise, create window, create context, etc
+    if(!Init())
+      return 3;
+
+	GLuint vao, vb, program;
+
+    void *contextA = mainContext;
+    void *contextB = MakeContext(mainWindow, NULL);
+
+    GLuint fboB, texB;
+    ActivateContext(mainWindow, contextB);
+    {
+      vao = MakeVAO();
```

Comment:
Unfortunately you won't be able to use the `Make*` functions as it stands with the unshared context. They push the resulting object into a list and delete it on `mainContext` before shutdown in `OpenGLGraphicsTest::Shutdown`.

It's possible you could add new `DeleteVAO` etc functions to remove the objects from the lists when manually deleting them on context B, but it's probably easier just to use `glGen` for everything.